src/pages/studyView/StudyViewPage.tsx

```diff
@@ -139,13 +142,29 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
 
         switch (chartMeta.chartType) {
             case ChartTypeEnum.PIE_CHART: {
-                props.promise = this.store.studyViewClinicalDataCountsCache.get({
-                    attribute: chartMeta.clinicalAttribute!,
-                    filters: this.store.filters
-                });
-                props.filters = this.store.getClinicalDataFiltersByUniqueKey(chartMeta.uniqueKey);
-                props.onValueSelection = this.handlers.onValueSelection;
-                props.onResetSelection = this.handlers.onValueSelection;
+
+                //custom charts
+                if (_.includes([UniqueKey.SAMPLES_PER_PATIENT, UniqueKey.WITH_CNA_DATA, UniqueKey.WITH_MUTATION_DATA], chartMeta.uniqueKey)) {
```

Comment:
give a name to that array of UniqueKeys so we can understand what we're matching against. and a comment here to explain